array-manipulation-3/src/flatten.js

```diff
@@ -0,0 +1,22 @@
+/* exported flatten */
+function flatten(array) {
+  var newArray = [];
+  for (var i in array) {
+    if (typeof array[i] === 'object') {
```

Comment:
I would use `Array.isArray()`, this would lead to false positives for actual `object`s